### PR TITLE
Add entitlement for Allowance and Allowance for Survivor

### PR DESCRIPTION
**Pull request by JeremyKennedy**
It's not perfect, but it's pretty close! Next todo is to add a "partner age" field, which should cover the biggest remaining edge case.

Outputted values confirmed good by Wiam/Joan.

### CODE REVIEW

File: utils/api/benefits/checkAfs.ts [hunk 1]
```diff
@@ -154,3 +159,28 @@ export default function checkAfs(params: CalculationInput): BenefitResult {
   // fallback
   throw new Error('should not be here')
 }
+
+class AfsEntitlement {
```
Comment (KhalidAdan) [inline utils/api/benefits/checkAfs.ts:171]: Nice work keeping these classes closely related! If you wanted extra nerd points, you could define an interface / type that enforces this so the developers that touch this after us don't unknowingly blow it up. 

100% not required though!

File: utils/api/benefits/checkAllowance.ts [hunk 2]
```diff
@@ -42,9 +44,12 @@ export default function checkAllowance(
     meetsReqPartner
   ) {
     if (meetsReqAge) {
+      const entitlementResult = new AllowanceEntitlement(
+        value.income
+      ).getEntitlement()
       return {
         eligibilityResult: ResultKey.ELIGIBLE,
-        entitlementResult: 0,
```
Comment (KhalidAdan) [inline utils/api/benefits/checkAllowance.ts:47]: 👨🏾‍🍳

**Review by KhalidAdan — COMMENTED** [review]
Let me know if you want me to take another look once it's not in draft!

File: utils/api/benefits/checkAfs.ts (hunk 1 above)
Comment (JeremyKennedy) [reply utils/api/benefits/checkAfs.ts:171]: I am pretty content with how quickly it breaks when you do something wrong thanks to the `OutputItemAfs` type (and the types for GIS and Allowance elsewhere), despite not using a base class here.

Always happy to get nerd points but pretty happy with it as is! 😃

**Comment by JeremyKennedy** [comment]
Added tests - ready for review!

File: __tests__/pages/api/index.test.ts [hunk 3]
```diff
@@ -1170,6 +1170,54 @@ describe('basic Allowance scenarios', () => {
...
+      partnerIncome: 0,
+      livingCountry: LivingCountry.CANADA,
+      legalStatus: LegalStatus.CANADIAN_CITIZEN,
+      yearsInCanadaSince18: 40,
+    })
+    expect(res.body.allowance.eligibilityResult).toEqual(ResultKey.ELIGIBLE)
+    expect(res.body.allowance.entitlementResult).toEqual(1206.41)
+    expect(res.body.allowance.reason).toEqual(ResultReason.NONE)
+  })
+})
```
Comment (KhalidAdan) [inline __tests__/pages/api/index.test.ts:1219]: 🎉

File: utils/api/scrapers/tsconfig.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
-    "esModuleInterop": true
+    "esModuleInterop": true,
+    "resolveJsonModule": true
```
Comment (KhalidAdan) [inline utils/api/scrapers/tsconfig.json:4]: Oh interesting, I didn't know about `resolveJsonModule`. TIL!

**Review by KhalidAdan — APPROVED** [review]
Looks good! Learned about some tsconfig stuff!

**Pull request merged**